src/commands/publish/preview/mod.rs

```diff
@@ -49,15 +49,25 @@ pub fn preview(
     };
 
     if livereload {
-        let ws_port: u16 = 8025;
+        let mut ws_port: u16 = 8025;
+
+        let server = loop {
+            let result = WebSocket::new(|out| FiddleMessageServer { out })?
+                .bind(format!("127.0.0.1:{}", ws_port)); //explicitly use 127.0.0.1, since localhost can resolve to 2 addresses
+
+            match result {
+                Ok(server) => break server,
+                //if 65535-8025 ports are filled, this will cause a panic due to the overflow
+                //if you are using that many ports, i salute you
```

Comment:
lol

Reply:
Interesting! I've never used websockets before, but I always thought a lot of these libraries would bake in their own random selection of an ephemeral port to use for egress traffic... I'm going to take a look :)

Reply:
Huh, looks like you do need to iterate until you find an available port! It is what it is :)

Reply:
Actually, could we add nice error handling that doesn't panic and throws a proper error if ws_port >= 65535 (max port)? We can even be fancy and ensure that this counter doesn't go over 49152 (IANA recommendation for bottom number of ephemeral port: https://en.wikipedia.org/wiki/Ephemeral_port)